rust/src/detect/transforms/mod.rs

```diff
@@ -29,29 +31,14 @@ pub mod strip_whitespace;
 pub mod urldecode;
 pub mod xor;
 
-#[repr(C)]
-#[allow(non_snake_case)]
-pub struct SCTransformTableElmt {
-    pub name: *const c_char,
-    pub desc: *const c_char,
-    pub url: *const c_char,
-    pub flags: u16,
-    pub Setup: unsafe extern "C" fn(de: *mut c_void, s: *mut c_void, raw: *const c_char) -> c_int,
-    pub Free: Option<unsafe extern "C" fn(de: *mut c_void, ptr: *mut c_void)>,
-    pub Transform: unsafe extern "C" fn(_det: *mut c_void, inspect_buf: *mut c_void, options: *mut c_void),
-    pub TransformValidate:
-        Option<unsafe extern "C" fn(content: *const u8, len: u16, context: *mut c_void) -> bool>,
-}
-
 /// cbindgen:ignore
 extern "C" {
     pub fn DetectSignatureAddTransform(
```

Comment:
Jason, Should this function be moved to (already bindgen'd) detect-engine-buffer.h (and prefixed with `SC` ) ?

Reply:
Yeah. IMO, If bindgen can do it or is already doing it, use the bindgen one and remove the hand coded binding.  Prefixing with SC is ideal, but not mandatory, can be attacked with another pass later.